From 22ba5662440d2865153ac1ff7de5f2b171414346 Mon Sep 17 00:00:00 2001
From: Sergey Kandaurov <pluknet@nginx.com>
Date: Thu, 10 Sep 2026 18:38:31 +0400
Subject: [PATCH] QUIC: ignore the QUIC extension received in TLS handshake

Instead of failing the handshake, it is now accepted and ignored like
any other unknown extension.  Unlike stated in RFC 9001, Section 8.2,
this represents less strict behaviour following RFC 8446, Section 4.2.
---
 src/event/quic/ngx_event_quic_openssl_compat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/event/quic/ngx_event_quic_openssl_compat.c b/src/event/quic/ngx_event_quic_openssl_compat.c
index 933cd7670f..dcdaa908f4 100644
--- a/src/event/quic/ngx_event_quic_openssl_compat.c
+++ b/src/event/quic/ngx_event_quic_openssl_compat.c
@@ -346,7 +346,7 @@ ngx_quic_compat_parse_transport_params_callback(SSL *ssl, unsigned int ext_type,
 
     c = ngx_ssl_get_connection(ssl);
     if (c->type != SOCK_DGRAM) {
-        return 0;
+        return 1;
     }
 
     ngx_log_debug0(NGX_LOG_DEBUG_EVENT, c->log, 0,
